### PR TITLE
Replace role system (RBAC) with permissions-based DB roles

### PR DESCRIPTION
**Pull request by AlanCoding**
##### SUMMARY
This replaces https://github.com/ansible/awx/pull/14735 as we have introduced a feature branch for this purpose.

This makes use of https://github.com/ansible/django-ansible-base/pull/45 and the intent is to have that merged before this.

New endpoints are exposed from DAB. These are
 - role definitions
 - role user assignments
 - role team assignments

The routers/views/serializers for these new endpoints live in DAB, not in this patch. This is mostly concerned with the "translation" layer to make the new roles look kind of like the old roles. Generally this is intended to be a backward-compatible change.

##### ISSUE TYPE
 - New or Enhanced Feature

##### COMPONENT NAME
 - API

### CODE REVIEW

File: awx/api/generics.py [hunk 1]
```diff
@@ -472,7 +476,11 @@ def skip_related_name(name):
 
 class ListCreateAPIView(ListAPIView, generics.ListCreateAPIView):
     # Base class for a list view that allows creating new objects.
-    pass
+    def perform_create(self, serializer):
+        super().perform_create(serializer)
+        if serializer.Meta.model in permission_registry.all_registered_models:
+            if self.request and self.request.user:
+                give_creator_permissions(self.request.user, serializer.instance)
```
Comment (fosterseth) [inline awx/api/generics.py:483]: trying to think of other resources that are created in non-conventional ways

ReceptorAddress in the create_or_update method of the InstanceSerializer
- I think this is admin-only anyways so maybe doesn't matter too much

BulkHost and BulkLaunch endpoints

Comment (AlanCoding) [reply awx/api/generics.py:483]: Good question. Those bulk endpoints create hosts & jobs. Hosts defer to inventory permission, and jobs defer to job template (or organization as fallback) permission. So no permission action is needed in those cases.

File: awx/api/views/__init__.py [hunk 2]
```diff
@@ -536,10 +540,12 @@ class InstanceGroupAccessList(ResourceAccessList):
 
 
 class InstanceGroupObjectRolesList(SubListAPIView):
+    deprecated = True
```
Comment (jbradberry) [inline awx/api/views/__init__.py:543]: Are we really going to deprecate everything?  That seems pretty hostile to community users.

(also, this one seems to have `deprecated = True` twice.)

Comment (PabloHiro) [reply awx/api/views/__init__.py:543]: Talking about deprecation, if I understood you correctly `access_list` should be marked as deprecated as well right?

Comment (AlanCoding) [reply awx/api/views/__init__.py:543]: Put up https://github.com/ansible/awx/pull/15068, which should address the things mentioned in both of the above 2 comments.

And yes, these will be deprecated. Obviously we won't just go break the UI. We will push the new UI to the community somehow and stop maintaining the old UI. Some time after that, yes, we will remove these endpoints, and we will delete the old `Role` tables that are used for the backward compatibility. Everything you could do with these you can do with the new endpoints.

**Review by PabloHiro — APPROVED** [review]

**Review by TheRealHaoLiu — APPROVED** [review]

**Review by fosterseth — APPROVED** [review]

**Review by chrismeyersfsu — APPROVED** [review]

**Pull request merged**